### PR TITLE
Cleans tensorParallelDegree with MultiDevice

### PR DESCRIPTION
**Pull request by zachgk**
This is a refactor to simplify the handling of tensor parallel degree. Before, it is read independently in 3+ locations in code and the behavior determining the tpDegree is hard to follow. This moves the reading to a single place and then represents it using a MultiDevice. This also means that the behavior can be entirely visible - a worker group will show up with the tp devices that it's worker will be using.

### CODE REVIEW

**Review by sindhuvahinis — APPROVED** [review]

**Comment by sindhuvahinis** [comment]
Now that we have MultiDevice, Here getVisibleDevices could make of use of this as well right? https://github.com/deepjavalibrary/djl-serving/blob/master/engines/python/src/main/java/ai/djl/python/engine/Connection.java#L207

**Comment by zachgk** [comment]
> Now that we have MultiDevice, Here getVisibleDevices could make of use of this as well right? https://github.com/deepjavalibrary/djl-serving/blob/master/engines/python/src/main/java/ai/djl/python/engine/Connection.java#L207

Yeah. The engine would already have the list of devices that should be part of the connection already. It doesn't use the `CUDA_VISIBLE_DEVICES` though.

Also, I talked with @frankfliu and we may save this PR for post-reinvent as it is hard to get the behavior right and we don't test every case even in integration. Although, I did add a bunch of tests which may be sufficient
